lib/manageiq/messaging/client.rb

```diff
@@ -161,18 +161,20 @@ def subscribe_background_job(options)
       end
 
       # Publish a message as a topic. All subscribers will receive a copy of the message.
-      # Expected keys in +options+ are:
+      # +messages+ can be either a hash or an array of hashes.
```

Comment:
Is it possible to encourage all messages to have the same general options?
so all messages have the same target and the same sender. Possibly the same headers, too?

Or will having multiple hashes trim down on having to `map` the incoming array and to multiple hashes inside this method?

Reply:
I like this idea a lot, in practice I think the options will all be the same and you'll just have an array of messages.

One of the nice parts of this approach though is that code written against the current interface will still work and MIQ isn't the only consumer of this API so this has the benefits of only enhancing and not breaking the current public API.

If we want to look at adding a different method or breaking the interface, splitting the options and payload would be really interesting